tftypes/attribute_path_test.go

```diff
@@ -585,3 +641,38 @@ func TestAttributePathString(t *testing.T) {
 		})
 	}
 }
+
+func attributePathStepComparer(i, j AttributePathStep) bool {
```

Comment:
I've been trying to implement an `Equal` method on these that `cmp` will pick up automatically, rather than defining comparers for them. Mostly because the comparers will need to be explicitly included everywhere, and it's hard to keep track of which you need. `Equal` works everywhere without needing to be explicitly included.

Reply:
Good to know that we're generally okay with expanding our compatibility surface area with `Equal()` methods. I think its low risk, but didn't want to presume.

Reply:
I think in this case it was a no-brainer to me because the interface was intentionally unfulfillable and was less of an interface and more of an enum. So consumers of the package have no real impact as a result of this change.

For other things I give it a bit more consideration, but even then I usually end up siding with including it (see the framework's `attr.Value` and `attr.Type`). No hard and fast rules, unfortunately. :(